tests/migrations/test_commands.py

```diff
@@ -1043,6 +1044,68 @@ class Meta():
+        msg_no_migrations = 'No migrations to prune.'
+        msg_must_remove_replaces_attribute = (
+            'Pruning cannot take place until the following squashed migrations '
+            "are recorded as applied (re-run 'manage.py migrate') "
+            'and have their replaces attribute removed:\n'
+        )
+        with self.temporary_migration_module(module='migrations.test_migrations_squashed') as migration_dir:
+            migration_dir_path = pathlib.Path(migration_dir)
+            squashed_migration_path = migration_dir_path.parent.name + '.migrations.0001_squashed_0002'
+            try:
+                call_command('migrate', 'migrations', verbosity=0)
+                # Delete the replaced migrations.
+                os.remove(migration_dir + '/0001_initial.py')
+                os.remove(migration_dir + '/0002_second.py')
+                # Attempt to prune before removing the "replaces" attribute.
+                with captured_stdout() as out:
+                    call_command('migrate', 'migrations', verbosity=1, prune=True)
+                out_value = out.getvalue()
+                self.assertIn(msg_must_remove_replaces_attribute, out_value)
+                self.assertIn('migrations.0001_squashed_0002', out_value)
+                self.assertIn(msg_no_migrations, out_value)
+                # Remove references to the deleted migrations from the squashed migration file.
+                with open(migration_dir + '/0001_squashed_0002.py', 'r') as f:
+                    data = f.read()
+                    data = data.replace('("migrations", "0001_initial"),', '')
+                    data = data.replace('("migrations", "0002_second"),', '')
+                    data = data.replace('replaces = [\n        \n        \n    ]', '')
+                with open(migration_dir + '/0001_squashed_0002.py', 'w') as f:
+                    f.write(data)
+                # Reload the module containing the squashed migration that was edited.
+                importlib.reload(sys.modules[squashed_migration_path])
+                # Prune.
+                with captured_stdout() as out:
+                    call_command('migrate', 'migrations', verbosity=1, prune=True)
+                self.assertIn(msg_prune, out.getvalue())
+                self.assertNotIn('Running migrations:', out.getvalue())
+                applied_migrations = recorder.applied_migrations()
+                self.assertEqual(
+                    {key for key in applied_migrations if key[0] == 'migrations'},
+                    {('migrations', '0001_squashed_0002')}
+                )
+                with captured_stdout() as out:
+                    call_command('migrate', 'migrations', verbosity=1, prune=True)
+                self.assertIn(msg_no_migrations, out.getvalue())
+            finally:
+                # Unmigrate everything.
+                call_command('migrate', 'migrations', 'zero', verbosity=0)
```

Comment:
IMO this test is too complicated because we're trying to covered several scenarios with a single test method. I'd split it into:
- `prune` with deleted squashed migrations (we can add a new set of migrations for this e.g. `test_migrations_squashed_no_replaces`),
- `prune` with deleted squashed migrations that are in `replaces`,
- `prune` when there is nothing to prune.